### PR TITLE
Update pdf2parquet to Docling v2

### PR DESCRIPTION
**Pull request by dolfim-ibm**
## Why are these changes needed?

The new Docling v2 allows to
1. Process more input formats: PDF, DOCX, PPTX, HTML, Markdown, ASCII Docs
2. Faster PDF backend
3. Improvements in the generated document
4. New DoclingDocument
5. Additional export format to plain text

In progress
- [x] upgrade dependencies and adapt code
- [x] add safe-guard for the download of model weights on multi-processes
- [x] implement flushing mechanism for creating batches of files
- [x] update doc_chunk tansform
- [x] propagate parameters to kfp

## Related issue number (if any).

- expose document hash of the input document. Refs https://github.com/IBM/data-prep-kit/issues/605
- use new faster PDF backend. Refs https://github.com/IBM/data-prep-kit/issues/573 
- safe-guard the download of model weights on multi-processes. Refs https://github.com/IBM/data-prep-kit/issues/667

### CODE REVIEW

**Comment by daw3rd** [comment]
I believe with the use of MultiLock this also fixes #667

File: transforms/language/doc_chunk/python/pyproject.toml
Comment (daw3rd) [inline transforms/language/doc_chunk/python/pyproject.toml:3]: not sure how we handle these version suffixes on transforms now.  But since you depend on a dev1 version of the core library, should this also be a dev1 suffix?

Comment (dolfim-ibm) [reply transforms/language/doc_chunk/python/pyproject.toml:3]: @touma-I which version should be used? I followed your advice here, but I can also change it.

File: transforms/language/doc_chunk/python/src/doc_chunk_transform.py
Comment (daw3rd) [inline transforms/language/doc_chunk/python/src/doc_chunk_transform.py:29]: I guess this is not backwards compatible with previous notebooks, scripts, etc?

File: transforms/language/pdf2parquet/python/requirements.txt
Comment (daw3rd) [inline transforms/language/pdf2parquet/python/requirements.txt:1]: Should this be >= 0.2.2.dev1.  dev1 may eventually go away.

File: transforms/language/pdf2parquet/python/src/pdf2parquet_transform.py
Comment (daw3rd) [inline transforms/language/pdf2parquet/python/src/pdf2parquet_transform.py:161]: Is it ok for artifacts_path to be None?  seems possible.

**Review by daw3rd — CHANGES_REQUESTED** [review]
Do any of. your tests exercise self.buffer and flush_binary()?

File: transforms/language/doc_chunk/python/pyproject.toml
Comment (touma-I) [reply transforms/language/doc_chunk/python/pyproject.toml:3]: @dolfim-ibm you are ok with what you have. The release process will fix all this...

File: transforms/language/doc_chunk/python/src/doc_chunk_transform.py
Comment (dolfim-ibm) [reply transforms/language/doc_chunk/python/src/doc_chunk_transform.py:29]: indeed. the parameter doesn't exist anymore in the library.
if you think it is better to keep it in the arguments and simply ignore it, we could add it again.

File: transforms/language/pdf2parquet/python/requirements.txt
Comment (dolfim-ibm) [reply transforms/language/pdf2parquet/python/requirements.txt:1]: here I rely on @daw3rd  and @touma-I expertise. let me know what is the correct value to use

File: transforms/language/pdf2parquet/python/src/pdf2parquet_transform.py
Comment (dolfim-ibm) [reply transforms/language/pdf2parquet/python/src/pdf2parquet_transform.py:161]: yes, it is actually the default value.

File: transforms/language/doc_chunk/python/src/doc_chunk_transform.py
Comment (daw3rd) [reply transforms/language/doc_chunk/python/src/doc_chunk_transform.py:29]: does this old parameter relate in anyway to the new parameters and could it be mapped somehow to the new parameter value (with a WARNING message of some sort).  Or, is this function no longer supported?  If the latter, then I would lean towards accepting it and warning about it so we don't break old notebooks?  I'm more worried about what example notebooks we have in the repo though, so if none use it, less of a problem for me to get rid of.

File: transforms/language/pdf2parquet/python/requirements.txt
Comment (daw3rd) [reply transforms/language/pdf2parquet/python/requirements.txt:1]: @touma-I I defer to you on this one.

File: transforms/language/doc_chunk/python/src/doc_chunk_transform.py
Comment (dolfim-ibm) [reply transforms/language/doc_chunk/python/src/doc_chunk_transform.py:29]: The parameter does not exists at all anymore. The new behavior, is sort of just covering what was needed.

On the other hand, for the notebook failures. Do DPK launchers complain if an argument is not existing? Or it is simply ignored?

Comment (dolfim-ibm) [reply transforms/language/doc_chunk/python/src/doc_chunk_transform.py:29]: I did a few tests (with and without). I think it should be covered now with the warning.

**Review by daw3rd — APPROVED** [review]
LGTM modulo @touma-I's call on the >= versioning.

File: transforms/language/pdf2parquet/python/requirements.txt
Comment (touma-I) [reply transforms/language/pdf2parquet/python/requirements.txt:1]: @dolfim-ibm @daw3rd this will be reset during the release process or whenever we push to pypi. for now, I wouldn't worry about it. You're good.

**Pull request merged**
